smrf/distribute/wind.py

```diff
@@ -590,6 +591,19 @@ def convert_wind_ninja(self, t):
         g_vel = utils.grid_interpolate(data_vel_int, self.vtx,
                                        self.wts, self.X.shape)
 
+        if np.any(np.isnan(g_vel)):
+            # https://stackoverflow.com/questions/9537543/replace-nans-in-numpy-array-with-closest-non-nan-value
+            self._logger.warning('NaN values present in the wind ninja interpolation, replaced with closest value')
+            for i in range(4):
+                mask = np.isnan(g_vel)
+                g_vel[mask] = np.interp(np.flatnonzero(mask), np.flatnonzero(~mask), g_vel[~mask])
```

Comment:
Interesting way of filling in the nan values, unfamiliar with the specifics of the method and it took me some digging to se how it would work. However, if, say, the left column is nan, it would use the second column values (closest elements) for the interpolation, but if it is filling in the last row(s) (if nans, as it's the problem being addressed at the moment), the flattened array would have nans for the last rows, and the closest values would be distant from the leftmost element of the array, and I wonder how that's actually working in the code using this.

for example:
a = [[0, 1, 2], [2, 3, 4], [np.nan, np.nan, np.nan]]
the last row would be interpolated from the y value for element in the fourth index of the flattened array (a.flat[4] that has a value of 4 in a).
- does the interpolation assign the value of a.flatten[4]  (4 in this case) to all the elements in the row? if so, is that what you intend? @scotthavens

Reply:
It's not a perfect solution to the problem as most interpolation methods won't extrapolate. This method works the best on the left columns as it will fill in with the nearest value. Like you pointed out, the bottom rows will be a single, repeated value, in your example 4. I didn't want to make the assumption that it was always the left and bottom values just in case.

I tried a few other interpolation techniques from `scipy.interpolate` and either they wouldn't extrapolate outside the domain or there were too many inputs points from the WindNinja outputs to interpolate.

Reply:
OK, sounds good. It is good to know what would happen on different locations of the matrix